packages/office-ui-fabric-react/src/utilities/DraggableZone/DraggableZone.tsx

```diff
@@ -0,0 +1,339 @@
+import * as React from 'react';
+import { IDraggableZoneProps, ICoordinates, IDragData } from './DraggableZone.types';
+import { getClassNames } from './DraggableZone.styles';
+import { BaseComponent } from '../../Utilities';
+
+export interface IDraggableZoneState {
+  isDragging: boolean;
+  position: ICoordinates;
+  lastPosition?: ICoordinates;
+}
+
+const eventMapping = {
+  touch: {
+    start: 'touchstart',
+    move: 'touchmove',
+    stop: 'touchend'
+  },
+  mouse: {
+    start: 'mousedown',
+    move: 'mousemove',
+    stop: 'mouseup'
+  }
+};
+
+// These are needed so that we can generalize the events
+// and so we have access to clientX and clientY in the touch events
+type MouseTouchEvent<T> = React.MouseEvent<T> & React.TouchEvent<T>;
+
+export class DraggableZone extends BaseComponent<IDraggableZoneProps, IDraggableZoneState> {
+  private _touchId?: number;
+  private _currentEventType = eventMapping.mouse;
+
+  constructor(props: IDraggableZoneProps) {
+    super(props);
+
+    this.state = {
+      isDragging: false,
+      position: this.props.position || { x: 0, y: 0 },
+      lastPosition: undefined
+    };
+  }
+
+  public componentWillReceiveProps(nextProps: IDraggableZoneProps) {
```

Comment:
componentWillRecieveProps is deprecated; you can instead use `getDerivedStateFromProps`.